zkvm/src/predicate.rs

```diff
@@ -32,8 +32,10 @@ pub enum PredicateWitness {
     /// (but known to be a valid Ristretto point).
     OpaqueBranch(RistrettoPoint),
 
-    /// Secret signing key for the predicate-as-a-verification-key.
-    Key(Scalar),
+    /// Signing key for the predicate-as-a-verification-key.
+    /// Prover will provide secret signing key separately when
+    /// constructing aggregated signature.
+    Key(RistrettoPoint),
```

Comment:
why not `Key(VerificationKey)`?

Reply:
We could do `Key(VerificationKey)` with a bigger refactor of `Predicate`, getting rid of the `PredicateWitness` type (which I think might be a good idea!), but right now we need to keep the key as uncompressed type to keep `to_uncompressed_point` unfailable

Reply:
ah, I spoke too soon! I think we might have to keep `Key(RistrettoPoint)` since we need the predicates to be uncompressed form for `to_point` to work correctly for `Predicate::Or` (pushed an updated commit)